gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApplicationMapper.java

```diff
@@ -178,6 +182,11 @@ public Application convert(ApplicationEntity applicationEntity, UriInfo uriInfo)
             application.setSettings(appSettings);
         }
 
+        if (applicationEntity.getApiKeyMode() != null) {
```

Comment:
If input applicationEntity.getApiKeyMode() is not set (let's say for example it's a consumer that directly call portal API without having added this data) ;
I think it's safer to let it null.

The ApplicationServiceImpl.checkApiKeyModeUpdate behave like that :
- For retro-compatibility : If input apiKey mode is not specified, get it from existing application
- If apiKeyMode is changed for example from SHARED to UNSPECIFIED : 400 error

So there is a risk to have this error here, trying to put it back to UNSPECIFIED

Reply:
This `convert` method is only used for GET calls. POST/PUT resources don't use it, so no risk here.
As seen we you, I mark the conversation as resolved.